include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -274,51 +284,46 @@ class fm_index
     //!\cond
         requires (text_layout_mode_ == text_layout::collection)
     //!\endcond
-    void construct(text_t && text)
+    void construct(text_t && text, bool reverse = false)
```

Comment:
why does this `construct` has a boolean for reversing in the normal `fm_index` when you have a whole class that can handle all the reversing?

Reply:
It just handles the reversing of the text; the alternative would be to copy the complete code of the construct method to the reverse_fm_index and add the tiny if I need...